### ISSUE TITLE
Net::LDAP is not mentioned as requirement

### ISSUE THREAD

**Issue by dilyanpalauzov** [cyrusimap/cassandane#73]
utils/fakeldapd needs Net::LDAP, but doc/README.deps does not mention it

**Comment by dilyanpalauzov** [comment]
Likewise, not mentioned and required are:
* Net::LDAP
* AnyEvent
* Convert::Base64
* DateTime::Format::ISO8601
* Net::LDAP::Server
* String::CRC32
* XML::DOM

Some of them, like AnyEvent are referenced in the botton for Fedora and Debian, but they are missing from the general section.

Since Net-DAVTalk requires Tie::DataUUID, XML::Spice and XML::Fast and is not installed over CPAN, the warnings from “perl Makefile.PL” are easily overseen.  Likewise for the stuff from Net::CalDAVTalk, Net::CardDAVTalk and Mail-JMAPTalk.

**Comment by elliefm** [comment]
Hmm, I guess this list is out of date too:
https://www.cyrusimap.org/imap/developer/developer-testing.html

**Comment by elliefm** [comment]
I think the web version is up to date now (or will be once the autoupdate hook runs), though it's hard to verify without setting up a blank environment and working through it step by step.

The *Talk modules are all available on CPAN these days (so their dependencies are fetched automatically).  The information saying to get them elsewhere is wrong/old, but I've updated it now.

**Comment by dilyanpalauzov** [comment]
I had to install all of the aforementioned modules from CPAN, so a blank evironment will need them, too.

**Comment by elliefm** [comment]
I've added them to the list but can't be bothered writing descriptions of them right now

**Comment by dilyanpalauzov** [comment]
Net::LDAP is not mentioned as requirement.

**Comment by dilyanpalauzov** [comment]
To be precise, the first error I got was, that Net::LDAP was not installed.  Then I installed it and was told, that Net::LDAP::Server is not installed.  As Net::LDAP::Server requires Net::LDAP, explicitly spelling the latter is in fact not necessary.

**Comment by elliefm** [comment]
Yeah, that's why I left it out.  It'd be nice to clean out all the ones (like Encode::IMAPUTF7) that are just dependencies of the other ones (and thus CPAN/apt/whatever will deal with the details) but that's an annoying rabbit hole for not a lot of real benefit, so I'm just gonna not stress about it instead.

But, thanks for your help updating this dependencies list.  It only really gets exercised thoroughly when someone sets up a new machine, so if we forget to update it as we introduce dependencies (which we clearly sometimes do) it gets out of date easily!  The feedback is appreciated :)

**Comment by dilyanpalauzov** [comment]
I propose substituting “2. Install dependencies” from cyrus-imap/docsrc/imap/developer/developer-testing.rst with “… as desrcibed in the doc/README.deps file” or doing anything else, which ensures that
* either the documentation is on one place,
* or all places documentig the requirements are kept synchronized.
